front_end/src/components/searchResultOverview/searchResultOverview.js

```diff
@@ -1,230 +1,325 @@
-import ReactStars from "react-stars";
+import { isWidthDown } from "@material-ui/core/withWidth";
 
-const styles = {
-  rating: { float: "left", width: "100%" },
-  imageStyle: { margin: 20, width: 200, height: 200, float: "left" }
-};
+// Component CSS to Javascript styles
+let styles = theme => ({
+  pageMargins: {
+    // Mobile margins
+    [theme.breakpoints.down("sm")]: {
+      marginLeft: "5%",
+      marginRight: "5%",
+      marginTop: "2%",
+      marginBottom: "2%"
+    },
+    // Tablet, laptop, desktop margins
+    [theme.breakpoints.up("md")]: {
+      marginLeft: "10%",
+      marginRight: "10%",
+      marginTop: "1%",
+      marginBottom: "1%"
+    }
+  },
+  subtitles: { fontWeight: "bold", color: "#808080" },
+  pad25: { padding: 25 }
+});
 
+// Main component of the Search Result Overview Page
 class searchResultOverview extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      // States used for sorting
+      sortCategory: "name",
+      sortOrder: "descending",
+
+      // States used for filter
+      star_rate: 0,
+      guest_rate: 0,
+      price_low: 0,
+      price_high: 0,
+      free_wifi: false,
+      free_parking: false,
+      free_breakfast: false,
+      pool: false,
+      pet_friendly: false
+    };
+
+    // Methods passed to child components
+    this.handleChange = this.handleChange.bind(this);
+    this.handleStarRatings = this.handleStarRatings.bind(this);
+    this.handleGuestRatings = this.handleGuestRatings.bind(this);
+    this.handleAmenities = this.handleAmenities.bind(this);
+    this.handlePriceRangeChange = this.handlePriceRangeChange.bind(this);
+    this.handleClickToHotel = this.handleClickToHotel.bind(this);
+  }
+
+  // Used to force user browser to scroll to top of page upon mounting this component
   componentDidMount = () => {
     window.scrollTo(0, 0);
   };
 
+  // Used to store the star rating input in <FiltersWindow />
+  handleStarRatings = newRating => {
+    this.setState({
+      star_rate: newRating
+    });
+  };
+
+  // Used to store the star rating input in <FiltersWindow />
+  handleGuestRatings = newRating => {
+    this.setState({
+      guest_rate: newRating
+    });
+  };
+
+  // Used to store the amenities input in <FiltersWindow />
+  handleAmenities = event => {
+    event.preventDefault();
+    this.setState({
+      [event.target.name]: event.target.checked
+    });
+  };
+
+  // Used to store the inputs from price range fields in <FiltersWindow />
+  handlePriceRangeChange = name => event => {
+    this.setState({
+      [name]: Number(event.target.value)
+    });
+  };
+
+  // Used to store the inputs from selections felds in <SortBar />
+  handleChange = event => {
+    event.preventDefault();
+    this.setState({
+      [event.target.name]: [event.target.value]
+    });
+  };
+
+  handleClickToHotel = hotel => event => {
+    event.preventDefault();
+    /*
+    // TODO: Update action call upon merging with backend
```

Comment:
Can you elaborate on this ?

Reply:
The backend API has changed. The values that will need to be passed into the action method needs to be updated based on the backend API.